fe/fe-core/src/main/java/com/starrocks/transaction/TransactionState.java

```diff
@@ -272,6 +272,36 @@ public String toString() {
     private long publishVersionTime = -1;
     private long publishVersionFinishTime = -1;
 
+    // The time of first commit attempt, i.e, the end time when ingestion write is completed.
+    // Measured in milliseconds since epoch.
+    // -1 means this field is unset.
+    //
+    // Protected by database lock.
+    //
+    // NOTE: This field is only used in shared data mode.
+    private long writeEndTimeMs = -1;
+
+    // The duration of the ingestion data write operation in milliseconds.
+    // This field is normally set automatically during commit based on
+    // writeEndTime and prepareTime. However, for cases like broker load
+    // with scheduling delays and concurrent ingestion, the auto calculated
+    // value may have large error compared to actual data write duration.
+    // In these cases, the upper ingestion job should set this field manually
+    // before commit.
+    //
+    // Protected by database lock.
+    //
+    // NOTE: This field is only used in shared data mode.
+    private long writeDurationMs = -1;
+
+    // The minimum time allowed to commit the transaction.
+    // Measured in milliseconds since epoch.
+    //
+    // Protected by database lock.
+    //
+    // NOTE: This field is only used in shared data mode.
+    private long allowCommitTimeMs = -1;
```

Comment:
Do you mean calculate this value each time `commit()` is called? If so, this may lead to starvation, for example, if transaction A is waiting but other transactions keep committing and increasing the compaction score, transaction A may never get to commit.